### PR TITLE
add custom relabel config while thanos storegateway hash shard

### PR DESCRIPTION
**Pull request by iyacontrol**
### Description of the change

add custom relabel config while thanos storegateway hash shard.

### Benefits

implement additional filtering functions, such as filtering by tenant or region.

### Possible drawbacks

no 

### Applicable issues

- #28633 

### Checklist

- [ ] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [ ] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [ ] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [ ] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/thanos/values.yaml [hunk 1]
```diff
@@ -3217,6 +3217,12 @@ storegateway:
   ## Sharded parameters
   ## @param storegateway.sharded.enabled Enable sharding for Thanos Store Gateway
   ## @param storegateway.sharded.hashPartitioning.shards Setting hashPartitioning will create multiple store statefulsets based on the number of shards specified using the hashmod of the blocks
+  ## @param storegateway.sharded.hashPartitioning.customRelabelConfig Setting custom relabel config
+  ## e,g:
+  ## customRelabelConfig: 
+  ##   - action: keep
+  ##     source_labels: ["region"]
+  ##     regex: cn-zhangjiakou
```
Comment (juan131) [inline bitnami/thanos/values.yaml:3225]: Should we name this parameter `extraRelabelingConfigs` instead? I think it'd better describe its purpose.

**Review by juan131 — COMMENTED** [review]
Thanks so much for this PR! Please check my comments

File: bitnami/thanos/values.yaml (hunk 1 above)
Comment (iyacontrol) [reply bitnami/thanos/values.yaml]: I am glad to see your response, and I have already submitted new changes as per your suggestions.

**Review by juan131 — APPROVED** [review]
LGTM

**Pull request merged**